xc7/utils/prjxray_db_cache.py

```diff
@@ -0,0 +1,53 @@
+"""
+This is a database cache. Its a kind of proxy object for Python sqlite3 connection
+which operates on a memory copy of the database.
+
+Upon object creation the database is "backed up" to memory. All subsequent
+operations are then pefromed on this copy which yields in performance increase.
+
+It is important to explicitly call the close() method once database connection
+is no longer needed. The method "backs up" the database back to disk.
+"""
```

Comment:
The typical solution to this problem is to use a context manager.  I suggest we take that approach here.

Reply:
I'd rather not implement `__enter__ / __exit__` as that would require wrapping the whole code into `with` statement. I can make a destructor `__del__` (which is I suppose how file objects implement closing).

Reply:
I made a destructor that does the writeback.

Reply:
Please implement __enter__ / __exit__, that is the way this should be handled.  __del__ has problems.  If you are concerned by the with statement, use an contextlib.ExitStack.

Reply:
Ok. So i'll do it in the `__enter__` / `__exit__` way

Reply:
Done.